### PR TITLE
 	Updated the DataCleanupTask and its scheduling mechanism

### PR DESCRIPTION
**Pull request by Ceiu**
- The DataCleanupTask now uses cron-like scheduling in place of the
  interval with units scheduling
- Events and compliance snapshots are now both cleaned up as part
  of the cleanup task
- Several tests added

### CODE REVIEW

File: gutterball/src/main/java/org/candlepin/gutterball/curator/ComplianceSnapshotCurator.java [hunk 1]
```diff
@@ -1149,5 +1157,52 @@ private int compareDatesByDate(Date date1, Date date2) {
...
+     * Deletes any compliances older than the specified duration in minutes.
+     *
+     * @param minutes
+     *  The number of minutes to use for cleaning up compliances
+     *
+     * @return
+     *  the number of compliance snapshots deleted during the cleanup operation
+     */
+    @Transactional
+    public int cleanupCompliances(int minutes) {
```
Comment (wottop) [inline gutterball/src/main/java/org/candlepin/gutterball/curator/ComplianceSnapshotCurator.java]: Does making this transactional mean that if there are a thousand compliances and one will not delete, none will?

**Comment by Ceiu** [comment]
Should. At least, that's the intent.
I should probably look into how @Transactional interacts with .flush so I can better answer your question.

File: gutterball/src/main/java/org/candlepin/gutterball/curator/ComplianceSnapshotCurator.java (hunk 1 above)
Comment (Ceiu) [inline gutterball/src/main/java/org/candlepin/gutterball/curator/ComplianceSnapshotCurator.java]: Should. At least, that's the intent.
I should probably look into @Transactional and how it interacts with .flush so I can better answer your question.

**Comment by wottop** [comment]
I am good with this whenever you get your quetions answered.

**Comment by Ceiu** [comment]
I made some changes after chatting with @nguyenfilip. Currently setting up to profile to make sure memory usage stays reasonable.

**Comment by Ceiu** [comment]
Alright, this should be the last version. Commits have been squashed, so I probably broke any checked out branches. Delete and try again, I suppose.

**Comment by wottop** [comment]
ack

**Pull request merged**